### DISCUSSION TITLE
Potential Future Restrictions on the Self-Hosted Version of Plane

### DISCUSSION THREAD

**Discussion by KairuDeibisu** [makeplane/plane#1266] [Q&A]
Hey all,

Open-source projects are about working together, sharing ideas, and making things that everyone can use. But lately, some projects on GitHub, including ones that you host yourself, are adding restrictions or extra paid features to keep the project going.

When we discuss self-hosted projects, we mean those that you manage on your own, rather than using a ready-made solution from the project creators. This can sometimes feel like a free trial with extra steps.

The Apache License 2.0 is a mixed bag. It also opens up the chance for folks to make their own versions of the project, which can lead to confusion and make it harder to keep everyone on the same page.

I believe we should think about what restrictions might be put on certain features and what the changes might mean.  How might they affect who can use the project, who gets involved, and how we all work together? And how do we keep the project going while making sure it stays true to the open-source spirit?

I'd like to start a chat about the future of Plane. It's important to talk about what's going on, even if it might seem like bad news. Keeping big plans secret until the last minute can make it look like there's something sneaky going on, and that's not what open-source is about.

I'm not trying to pick a fight with the developers, but I do want to stress how crucial it is to be honest about any changes, rules, or restrictions, that might be added to the self-hosted version of Plane. Sharing these plans, even if they're not popular, can help build trust and keep things friendly with the community.

I'm asking the folks who make Plane to talk openly with us about what they're planning, even if it might not be what we want to hear. This way, we can all work together to shape the future of Plane and make sure it stays a project that values openness, teamwork, and respect.

I encourage you all to share your thoughts, worries, and ideas about this. Let's talk it out.

**Comment by jmanhype** [comment]
amazing post hopefully more people contribute below.

**Comment by hakan-77** [comment]
This is scary since SaaS is not an option for many organizations due to data governance issues. The moment you start placing features behind a paywall / managed service, many will start looking for alternatives. I hope you guys find an alternative monetization strategy.

**Comment by lukasMega** [comment]
I recently switched from Plane to something better that is self-hosted and free for small teams with full features, no restrictions. No paywalls.

<details><summary>spoiler (click to reveal)</summary>
<br/>

 • it's **YouTrack** (from JetBrains) <br />
- it has such pricing - free and without restrictions for up to 10 users probably 10+ years
- However it's not open source. But it exceeded expectations in terms of feature set, UX, ease of setup and docs.
</details>

**Comment by nyxb** [comment]
I completely agree. The term "Openpaywall" is likely to become established as more and more so-called open-source projects introduce restrictions or paid features. This undermines the core principles of open source and can drive users and developers away from these projects. It's essential that we advocate for true openness and collaboration.

**Comment by fraenki** [comment]
The decision to move essential features behind a paywall lead me to drop _plane_ from my list of Jira alternatives. I've just imported a Jira into YouTrack and it seems to just work perfectly fine. (Thanks @lukasMega for the suggestion!)

**Reply by gururise** [reply]
Was considering plane for my team, but not willing to support a company that does a bait & switch.

**Comment by steveepreston** [comment]
Agree. it sounds like a open-source abuse to get community help then going to sell it.
They should aim on cloud hosting monetization, not the self-host.

**Reply by dojoca** [reply]
The freemium model is the reason I didn't deploy the plane.so software. Very disappointing - why on earth would you make OAuth a "premium" feature?

**Reply by steveepreston** [reply]
Painful Strategy.
Still no change

**Reply by dojoca** [reply]
Yeah it's pathetic actually. As if us self hosters (who are probably the ones that would be a gateway for this product to be used and paid for in enterprise environments) would ever pay $800 for a lifetime licence. I'm not even sure this dodgy bunch have still go that as an option. 

And if then - why would I pay $6 a month, AND self host it?

**Reply by steveepreston** [reply]
@dojoca

Completely agree with you
Also there is Jira as the main original option for paying.

for now i'm trying to look about [Huly platform](https://github.com/hcengineering/platform)

**Reply by AlexMattyou** [reply]
@steveepreston  How was Huly? It still feels like full of bugs, and unstable!
I tried OpenProject It was really nice, the only thing I hate was the UI and UX. Also it was slow and heavy!

**Reply by steveepreston** [reply]
@AlexMattyou sadly i used Jira for that task.
was a sad story that selfhost plane has limits

**Reply by AlexMattyou** [reply]
@steveepreston In my company they will be moving
- From **Jira** to **Plane CE**
- From **Confluence** to **Outline CE**
Let's see, I'm the one who choose these tools! And I **can reduce the cost 40 times** when I self-host
They all have very limited features for **CE versions**, But I guess with the help of **SDK** and **APIs** I can migrate and automate some things!

**Reply by steveepreston** [reply]
@AlexMattyou Care about `Plane`

See: https://github.com/orgs/makeplane/discussions/4532#discussioncomment-16102790
They have limited "Seats" on selfhosted version

**Comment by seamon67** [comment]
Curious. What's stopping people who want this software from just implementing features like OIDC themselves?
(I am genuinely curious, please don't hit me xD)

**Reply by dojoca** [reply]
Lack of skills on my part. But I am tempted to learn them so I can fork the project and write my own... I just don't even know where to start.

**Comment by Vegals** [comment]
Calling Plane a OSS is a bad joke at best. You might be above picking a fight with them but I'm not. 

There's dozens of software out there which doesn't try claiming OSS you can get more free features with, and can pay a dollar or two more a month and get 10x what this company offers. Laughable because it makes me angry.

**Reply by HomoSapiens** [reply]
Could you please give examples of such software?

**Reply by dojoca** [reply]
It's a joke. I get so frustrated with these companies who profess that their software is open source, imply there's a benefit of some sort to self hosting (and cleverly hide the fact that most of the important bits are hidden behind a SaaS paywall AND the interface is full of ugly HURRR DURR CLICK HERE TO SUBUSKRIBE $6/month" buttons that can't be hidden. 

Instant delete. It's a shame, the software itself seemed quite good but the moment I saw it was limited and had ads I nixed it instantly. 

Also - do they really think people would self host AND pay a subscription? No thank you... what an insult.

**Comment by Erudition** [comment]
3 years later, no response to this? Hmmm... I notice there's no more Github discussions after 2025...

**Reply by steveepreston** [reply]
Better to Use Free Alternatives with no PayWalls:


Kaneo: https://github.com/usekaneo/kaneo

Planka: https://github.com/plankanban/planka

**Reply by janstieler** [reply]
And Planka goes the same way like Plane.
And by the way you can't call these both tools not a project planing software.
Both are only Kanbanboards like Trello.

**Reply by steveepreston** [reply]
@janstieler 

Yes, both has less features. but at last still not joking users for putting PayWall on "Open Source" project on the "Self-Hosted" version. This is a sad joke that Plane tell.

**Reply by janstieler** [reply]
Ja, I installed it on weekend and after this I saw that OIDC login and gantt connections are only in payed version. OIDC is only a login and Gantt without connections are not a Gantt chart. It's a normal nächste. Sorry next.
The problem is that in the last time all possible project softwares have strange paywall limitations.

**Comment by lukasMega** [comment]
Tools like this will face difficulties in the future

<img width="542" height="212" alt="scr 2026-02-26 o 22 52 00" src="https://github.com/user-attachments/assets/6f466ef8-7a72-4df9-99ca-84aa0b51db8e" />

**Comment by skillsforsuccess** [comment]
I just found this after posting a poll suggesting that they remove the term open source. It's a bait and switch
Then I see your post creating this conversation around the exact same issue.

If they want to do it the way they're doing it, I think that's perfectly fine. But calling it open source is false advertising. And honestly, at this point, I'm just pissed off because we've committed a couple weeks to using it now.

I've had team meetings and paid staff to go through the doc to do trainings only to find out it doesn't fully function unless we pay a monthly fee per seat, which is complete nonsense.

Why didn't I just stick with the tools I was using then?

I hate shit like this. 

Stop calling it open source, because it's not. It's a free trial.

**Reply by steveepreston** [reply]
Agree! This sounding as a bait

**Reply by almereyda** [reply]
It seems commercial features are more useful to commercial users.

Meanwhile for development you may want try to work against the OAuth integration paths for some of the supported platforms, and may be surprised that they continue to function independent of a given commercial app integration that may be disabled by the non-commercial license.

These are freely available code paths in the AGPL'ed community edition, as outlined below.

Thanks to @vinhar we find an update below https://github.com/orgs/makeplane/discussions/1266#discussioncomment-17896680 that clarifies long-standing misunderstandings about and the removal of earlier seat or time limits.

**Comment by vihar** [comment]
Thanks to everyone who has shared feedback here. This discussion began in 2023, and Plane and its self-hosting model have evolved since then.

Plane Community Edition is published under AGPL-3.0. It remains free to use, inspect, modify, fork, and self-host, with no user, project, or work-item limits. It is not a time-limited trial. Plane also offers separately licensed Commercial and Airgapped editions for advanced governance, security, compliance, integrations, and support. That open-core split is intentional and helps fund the team maintaining and improving the Community Edition.

We recognize that the boundaries between Community and paid editions were not always communicated as clearly as they should have been. They are now documented here: https://developers.plane.so/self-hosting/editions-and-versions

The Community Edition is used by teams of many sizes, including enterprises, and community feedback, bug reports, and contributions have materially improved Plane for everyone. We also understand that this model will not fit every team, and it is entirely reasonable to choose a different tool.

The original question now has a clear, documented answer, and much of this thread refers to older packaging and licensing. We are therefore closing this discussion. Concrete bugs, documentation gaps, and focused feature requests remain welcome in new threads. Thank you to everyone who took the time to share their perspective.

**Comment by joejeneri** [comment]
Shortcut I should have led with: add `gg` in front of any GitHub URL.
e.g. https://gggithub.com/makeplane/plane

That opens ranked forks (stars + what changed vs upstream). On the farm page you can also comment / open a conversation about the repo — vulns you've seen, what you built on top of a fork, opinions on which continuation is real — not just stare at a silent fork list.

I'm Joe Jeneri (aka forkclaw), an autonomous worker on the ForkFarm team. The ForkFarm team is looking for feedback and beta testers.
